src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -550,208 +552,213 @@ namespace DataZoneEquipment {
-                ZoneEquipList(ControlledZoneNum).HeatingCapacity.allocate(ZoneEquipList(ControlledZoneNum).NumOfEquipTypes);
-                ZoneEquipList(ControlledZoneNum).EquipType = "";
-                ZoneEquipList(ControlledZoneNum).EquipType_Num = 0;
-                ZoneEquipList(ControlledZoneNum).EquipName = "";
-                ZoneEquipList(ControlledZoneNum).EquipIndex = 0;
-                ZoneEquipList(ControlledZoneNum).CoolingPriority = 0;
-                ZoneEquipList(ControlledZoneNum).HeatingPriority = 0;
-                ZoneEquipList(ControlledZoneNum).CoolingCapacity = 0;
-                ZoneEquipList(ControlledZoneNum).HeatingCapacity = 0;
-
-                for (ZoneEquipTypeNum = 1; ZoneEquipTypeNum <= ZoneEquipList(ControlledZoneNum).NumOfEquipTypes; ++ZoneEquipTypeNum) {
-                    ZoneEquipList(ControlledZoneNum).EquipType(ZoneEquipTypeNum) = AlphArray(2 * ZoneEquipTypeNum + 1);
-                    ZoneEquipList(ControlledZoneNum).EquipName(ZoneEquipTypeNum) = AlphArray(2 * ZoneEquipTypeNum + 2);
-                    ValidateComponent(ZoneEquipList(ControlledZoneNum).EquipType(ZoneEquipTypeNum),
-                                      ZoneEquipList(ControlledZoneNum).EquipName(ZoneEquipTypeNum),
+                thisZoneEquipList.NumOfEquipTypes = maxEquipCount;
+                thisZoneEquipList.EquipType.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.EquipType_Num.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.compPointer.resize(thisZoneEquipList.NumOfEquipTypes + 1);
+                thisZoneEquipList.EquipName.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.EquipIndex.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.EquipData.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.CoolingPriority.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.HeatingPriority.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.CoolingCapacity.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.HeatingCapacity.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.SequentialCoolingFraction.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.SequentialHeatingFraction.allocate(thisZoneEquipList.NumOfEquipTypes);
+                thisZoneEquipList.EquipType = "";
+                thisZoneEquipList.EquipType_Num = 0;
+                thisZoneEquipList.EquipName = "";
+                thisZoneEquipList.EquipIndex = 0;
+                thisZoneEquipList.CoolingPriority = 0;
+                thisZoneEquipList.HeatingPriority = 0;
+                thisZoneEquipList.CoolingCapacity = 0;
+                thisZoneEquipList.HeatingCapacity = 0;
+
```

Comment:
`SequentialCoolingFraction` and `SequentialHeatingFraction` should be initialized to 1.0 here, similar to priority and capacity.